controllers/openstackdataplanenode_controller.go

```diff
@@ -225,12 +243,15 @@ func (r *OpenStackDataPlaneNodeReconciler) Reconcile(ctx context.Context, req ct
 		r.Log.Info("Set DeploymentReadyCondition true", "instance", instance)
 		instance.Status.Conditions.Set(condition.TrueCondition(condition.DeploymentReadyCondition, condition.DeploymentReadyMessage))
 
-		// Explicitly set instance.Spec.Deploy = false
-		// We don't want another deploy triggered by any reconcile request, it
-		// should only be triggered when the user (or another controller)
-		// specifically sets it to true.
-		instance.Spec.DeployStrategy.Deploy = false
-
+		// Explicitly set instance.Status.Deploy = false
+		// Explicitly set instance.Status.DeployIdentifier = ""
+		// We don't want another deploy triggered by any reconcile request, it should
+		// only be triggered when the user (or another controller) specifically
+		// sets it to true.
+		r.Log.Info("Set Deploy to false")
+		instance.Status.Deploy = false
+		r.Log.Info("Set DeployIdentifier to empty value")
+		instance.Status.DeployIdentifier = ""
```

Comment:
is the switch to setting these on Status since we couldn't figure out how to update the Spec? I guess my concern with this is that Spec.DeployStrategy.Deploy would still be true after a deploy.

Then the status fields are set to false. So on the next reconcile wouldn't it start a new deploy?

To edit the fields on the spec can we use Update as seen here?
https://cloud.redhat.com/blog/kubernetes-operators-best-practices
(search for r.GetClient().Update)

Reply:
You're right, it seems that it was working but it keep updating ansibleee CRs even if their done. I'll try again with `r.GetClient().Update` approach.